flask-api/flaskr/candidate.py

```diff
@@ -0,0 +1,201 @@
+from flask import Blueprint, abort, jsonify, request, make_response
+from flaskr.db import get_db
+from flask_jwt_extended import jwt_required
+import json
+
+bp = Blueprint('candidates', __name__, url_prefix='/candidates')
+
+@bp.route('/create', methods=['POST'])
+@jwt_required
+def create():
+    if not(request.data):
+        abort(400, 'Bad Request')
+
+    try:
+        candidate_name = request.json['candidate_name']
+        v_event_id = request.json['v_event_id']
+    except:
+        abort(400, 'Bad Request')
+
+    try:
+        party_id = request.json['party_id']
+    except:
+        party_id = None
+
+    db = get_db()
+
+    # Check if the Party exists
+    if (party_id != None):
+        party = db.execute(
+            'SELECT * FROM party WHERE id = ?', (party_id,)
+        ).fetchone()
+
+        if party is None:
+            abort(400, 'Bad Request')
+
+    # Check if the Voting Event exists
+    v_event = db.execute(
+        'SELECT * FROM voting_event WHERE id = ?', (v_event_id,)
+    ).fetchone()
+
+    if v_event is None:
+        abort(400, 'Bad Request')
+
+    # Save to DB
+    db.execute(
+        'INSERT INTO candidate (candidate_name, party_id, v_event_id) VALUES (?, ?, ?)',
+        (candidate_name, party_id, v_event_id)
+    )
+    db.commit()
+
+    return '', 204
+
+@bp.route('/<candidate_id>', methods=['GET'])
+@jwt_required
+def get(candidate_id):
+    db = get_db()
+    cursor = db.cursor()
+    candidate = cursor.execute(
+        'SELECT * FROM candidate WHERE id = ?', (candidate_id,)
+    ).fetchone()
+
+    if candidate is None:
+        abort(404, 'Not Found')
+
+    # Serialise response into a JSON object
+    data = {}
+    for key in candidate.keys():
+        data[key] = candidate[key]
+
+    resp = make_response(json.dumps(data, default=str), 200)
+    resp.headers['Content-Type'] = 'application/json'
+    return resp
+
+@bp.route('', methods=['GET'])
+@jwt_required
+def getList():
+    candidate_name = request.args.get('candidate_name', None)
+    party_id = request.args.get('party_id', None)
+    v_event_id = request.args.get('v_event_id', None)
+
+    db = get_db()
+    cursor = db.cursor()
+
+    if (candidate_name == None and party_id == None and v_event_id == None):

```

Comment:
This part may look a bit kekw, but I don't know how to simplify it. Since it uses prepared statements, I can't just glob it since it's sanitised.